lib/advert/ui/components/association_bar.dart

```diff
@@ -52,11 +52,33 @@ class AssociationBar extends HookConsumerWidget {
               },
               associationId: e.id,
               name: e.name,
-              avatarName: e.name
-                  .split(' ')
-                  .take(2)
-                  .map((s) => s[0].toUpperCase())
-                  .join(),
+              avatarName: () {
+                try {
+                  if (e.name.length <= 3) {
+                    return e.name;
+                  } else if (e.name.contains(" ")) {
+                    return e.name
+                        .split(" ")
+                        .where((s) => s.isNotEmpty)
+                        .take(2)
+                        .map((s) => s[0].toUpperCase())
+                        .join();
+                  } else if (e.name.contains("'")) {
+                    return e.name
+                        .split("'")
+                        .where((s) => s.isNotEmpty)
+                        .take(2)
+                        .map((s) => s[0].toUpperCase())
+                        .join();
```

Comment:
```
return e.name
                        .split(" ")
                        .where((s) => s.isNotEmpty)
                        .take(2)
                        .map((s) => s[0].toUpperCase())
                        .join();
                  } else if (e.name.contains("'")) {
                    return e.name
                        .split("'")
                        .where((s) => s.isNotEmpty)
                        .take(2)
                        .map((s) => s[0].toUpperCase())
                        .join();
```

Split then flatten then resplit, way cleaner. And don't forget to trim before any acction